File: snippets/swift/BreezSDKExamples/Sources/Webhook.swift
//
//  ServiceStatus.swift
//  
//
//

import Foundation
import BreezSDKLiquid

func registerWebhook(sdk: BindingLiquidSdk) throws {
    // ANCHOR: register-webook
    try sdk.registerWebhook(webhookUrl: "https://your-nds-service.com/notify?platform=ios&token=<PUSH_TOKEN>")  
    // ANCHOR_END: register-webook
}

func unregisterWebhook(sdk: BindingLiquidSdk) throws {
    // ANCHOR: unregister-webook
    try sdk.unregisterWebhook()  
    // ANCHOR_END: unregister-webook
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/Webhook.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*
class Webhooks {
    fun registerWebhook(sdk: BindingLiquidSdk) {
        // ANCHOR: register-webook
        try {
            sdk.registerWebhook("https://your-nds-service.com/api/v1/notify?platform=ios&token=<PUSH_TOKEN>")
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: register-webook
    }

    fun unregisterWebhook(sdk: BindingLiquidSdk) {
        // ANCHOR: unregister-webook
        try {
            sdk.unregisterWebhook()
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: unregister-webook
    }
}